lib/AST/ASTWalker.cpp

```diff
@@ -455,20 +455,20 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
       else
         return true;
     }
+
     bool alreadyFailed = false;
     if (shouldWalkExpansion) {
       MED->forEachExpandedNode([&](ASTNode expandedNode) {
         if (alreadyFailed) return;
+
         if (auto *expr = expandedNode.dyn_cast<Expr *>()) {
-          if (!doIt(expr))
-            alreadyFailed = true;
+          alreadyFailed = doIt(expr);
```

Comment:
@hamishknight do you have the refactoring itch again by any chance :)?

Reply:
Heh, maybe one day, also need to do SourceEntityWalker :)

Reply:
So. Fun story. These were actually correct. They were checking for `nullptr`, not for bool. `doIt(Decl*)` returns true on failure. The others return `nullptr`. Yay implicit conversions.